File: Object oriented programming (II sem)/OOP in java/Turtle.java
import java.awt.*;

public class Turtle extends Animal {
    public Turtle(Position position, World world){
        super("Turtle", 2, 1,new Color(67, 112, 0, 255), position, world);
    }

    @Override
    public void action(){
        if(world.chanceOfBeingTrue(0.25f)) super.action();
    }

    @Override
    public boolean blockedAttack(Organism other){
        return other.getStrength() < 5;
    }
}

File: Object oriented programming (II sem)/OOP in cpp/Turtle.cpp
#include"Turtle.h"

Turtle::Turtle(Position pos, World& world) :
	Animal(TURTLE_NAME, TURTLE_STRENGTH, TURTLE_INITIATIVE, TURTLE_ICON, pos, world) {}

void Turtle::action() {
	if (world.chanceOfBeeingTrue(0.75f)) {
		return;
	}
	Position newPos = pickRandomNeighbourField(STANDARD_RANGE);
	if (world.isEmpty(newPos)) {
		move(newPos);
	}
	else {
		collision(world.getBoard()[newPos.x][newPos.y]);
	}
}

void Turtle::collision(Organism* other) {

}